lib/macho/fat_file.rb

```diff
@@ -312,6 +329,33 @@ def get_machos
       machos
     end
 
+    # Yield each Mach-O object in the file, rescuing and accumulating errors.
+    # @param options [Hash]
+    # @option options [Boolean] :strict (true) whether or not to fail loudly
+    #  with an exception if at least one Mach-O raises an exception. If false,
+    #  only raises an exception if *all* Mach-Os raise exceptions.
+    # @raise [MachO::MachOSliceModificationError] under the conditions of
+    #  the `:strict` option above.
+    # @api private
+    def each_macho(options = {})
+      errors = []
+
+      machos.each_with_index do |macho, index|
+        begin
+          yield macho
+        rescue MachOModificationError => e
+          errors << [index, e]
+        end
+      end
+
+      strict = options.fetch(:strict, false)
```

Comment:
This disagrees with the documentation regarding the default for `:strict`. However, I think the default suggested by the documentation is ultimately the better choice (i.e. we should require an opt-out of the strict behavior) and for this reason, this should become `options.fetch(:strict, true)`.